lib/features/thread/presentation/thread_controller.dart

```diff
@@ -3,57 +3,172 @@ import 'package:core/presentation/utils/responsive_utils.dart';
+    mailboxDashBoardController.selectedMailbox.listen((selectedMailbox) {
+      mailboxDashBoardController.setSelectedEmail(PresentationEmail.presentationEmailEmpty);
+      refreshGetAllEmailAction();
+    });
   }
 
-  SelectMode getSelectMode(PresentationThread presentationThread) {
-    return presentationMailCurrent.value.id == presentationThread.id
-      ? SelectMode.ACTIVE
-      : SelectMode.INACTIVE;
+  @override
+  void dispose() {
+    super.dispose();
+    mailboxDashBoardController.selectedMailbox.close();
   }
 
-  void selectMail(BuildContext context, PresentationThread presentationMailSelected) {
-    presentationMailCurrent.value = presentationMailSelected;
-    mailboxDashBoardController.mailCurrent.value = presentationMailSelected;
+  @override
+  void onData(Either<Failure, Success> newState) {
+    super.onData(newState);
+  }
 
-    if (!responsiveUtils.isDesktop(context)) {
-      goToMail(context);
+  @override
+  void onDone() {
+    viewState.value.fold(
+      (failure) => _resetPositionCurrentAndLoadMoreState(),
+      (success) => _updateEmailList(success is GetAllEmailSuccess ? success.emailList : [])
+    );
+  }
+
+  @override
+  void onError(error) {
+    _resetPositionCurrentAndLoadMoreState();
+  }
+
+  void _updateEmailList(List<PresentationEmail> newListEmail) {
+    emailList.value += newListEmail;
+
+    lastGetTotal = emailList.length;
+
+    if (newListEmail.isEmpty) {
+      _resetPositionCurrentAndLoadMoreState();
+    } else {
+      loadMoreState.value = LoadMoreState.IDLE;
     }
   }
 
-  void goToMailbox({required GlobalKey<ScaffoldState> keyWidgetMailboxContainer}) {
-    keyWidgetMailboxContainer.currentState?.openDrawer();
+  EmailFilterCondition? _getFilterConditionCurrent() {
+    return mailboxDashBoardController.selectedMailbox.value.id != PresentationMailbox.presentationMailboxEmpty.id
+      ? EmailFilterCondition(inMailbox: mailboxDashBoardController.selectedMailbox.value.id)
+      : null;
   }
 
-  void goToMail(BuildContext context) {
-    Get.toNamed(AppRoutes.MAIL);
+  Set<Comparator>? _getSortCurrent() {
+    return Set()
+      ..add(EmailComparator(EmailComparatorProperty.sentAt)
+        ..setIsAscending(false));
   }
 
-  @override
-  void onData(Either<Failure, Success> newState) {
-    super.onData(newState);
+  void _resetPositionCurrentAndLoadMoreState() {
+    if (loadMoreState.value == LoadMoreState.LOADING) {
+      positionCurrent -= lastGetTotal;
+    }
+    loadMoreState.value = LoadMoreState.COMPLETED;
   }
 
-  @override
-  void onDone() {}
+  void getAllEmailAction(AccountId accountId,
+    {
+      UnsignedInt? limit,
+      int position = 0,
+      Set<Comparator>? sort,
+      Filter? filter
+    }
+  ) async {
+    consumeState(_getEmailsInMailboxInteractor.execute(
+      accountId,
+      limit: limit,
+      position: position,
+      sort: sort,
+      filter: filter,
+      properties: Email.allProperties
```

Comment:
Please note: only get what you need, I am not agreed to using `allProperties` here. Please define a static for Thread only.

Reply:
allProperties is just including the properties described as the api example in this PR `https://github.com/linagora/tmail-flutter/issues/6`. Please read it.

Reply:
Yes, I know, but it is not safe for us, when the library change `allProperties`. So I want to create separate properties for this get.

Reply:
I think you can dis-ambiguate that by calling this constant `allListingProperties`